### DISCUSSION TITLE
Dataframe to be big to be shown properly?

### DISCUSSION THREAD

**Discussion by Reapor-Yurnero** [ismms-himc/clustergrammer2#89] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/15375211/108291989-12af6280-7148-11eb-85d7-3f6e05ab2a3a.png)
It seems like the dataframe is too big and nothing is successfully rendered? Does anyone know how to solve this?

**Comment by cornhundred** [comment]
Hi @Reapor-Yurnero Clustergrammer2 should be able to handle that size of a matrix. You can try with a smaller subset of the matrix first by doing 
```
net.load_df(euc_df.iloc[0:100,0:100])
net.widget()
```

**Comment by Reapor-Yurnero** [comment, marked as answer]
@cornhundred I‘ve figured this out. It seems my dataframe is having index of dtype "Int64" and can't be rendered by Clustergrammer2 properly. I changed to normal python object dtype and it works. But I believe this should not be the expected behavior so probably there's a bug behind it that you may want to fix

**Reply by cornhundred** [reply]
Thanks @Reapor-Yurnero, yes we should fix this behavior. Thank you for pointing this out.

**Reply by Reapor-Yurnero** [reply]
FYI, a very simple way to reproduce this. When loading df from csv, the index would be of int64 type rather than str object, and the rendering would thus fail.
```python
import numpy as np
import pandas as pd

# generate random matrix
num_rows = 100
num_cols = 100
np.random.seed(seed=100)
mat = np.random.rand(num_rows, num_cols)

# make row and col labels
rows = range(num_rows)
cols = range(num_cols)
rows = [str(i) for i in rows]
cols = [str(i) for i in cols]

# make dataframe 
df1 = pd.DataFrame(data=mat, columns=cols, index=rows)
print(df1.dtypes)

df.to_csv('test.csv')
df2 = pd.read_csv('test.csv', header=0,index_col=0)
# df2.index = df2.index.map(str) # without this line, rendering below is not possible
print(df2.dtypes)

from clustergrammer2 import net

net.load_df(df2)
net.cluster()
net.widget()
```
